homeassistant/components/nexia/climate.py

```diff
@@ -318,118 +314,163 @@ def set_temperature(self, **kwargs):
             if new_cool_temp - new_heat_temp < deadband:
                 new_heat_temp = new_cool_temp - deadband
 
-        self._device.set_heat_cool_temp(
+        self._zone.set_heat_cool_temp(
             heat_temperature=new_heat_temp,
             cool_temperature=new_cool_temp,
             set_temperature=set_temp,
         )
-        self.schedule_update_ha_state()
+        self._signal_zone_update()
 
     @property
     def is_aux_heat(self):
         """Emergency heat state."""
-        return self.thermostat.is_emergency_heat_active()
-
-    @property
-    def device_info(self):
-        """Return the device_info of the device."""
-        return {
-            "identifiers": {(DOMAIN, self._device.zone_id)},
-            "name": self._device.get_name(),
-            "model": self.thermostat.get_model(),
-            "sw_version": self.thermostat.get_firmware(),
-            "manufacturer": MANUFACTURER,
-            "via_device": (DOMAIN, self.thermostat.thermostat_id),
-        }
+        return self._thermostat.is_emergency_heat_active()
 
     @property
     def device_state_attributes(self):
         """Return the device specific state attributes."""
-        data = {
-            ATTR_ATTRIBUTION: ATTRIBUTION,
-            ATTR_ZONE_STATUS: self._device.get_status(),
-        }
+        data = super().device_state_attributes
+
+        data.update({ATTR_ZONE_STATUS: self._zone.get_status()})
```

Comment:
This will be faster.

```suggestion
        data[ATTR_ZONE_STATUS] = self._zone.get_status()
```

Reply:
Thanks. Much cleaner.